From 603df3b608406f7db2346c4a7cdb6a3c889bfa0f Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 13 Sep 2024 16:08:23 +0200
Subject: [PATCH 1208/1695] module: Don't consider OP-TEE modules when
 SCP_OPTEE_DIR is not set

Do not consider OP-TEE modules when SCP_OPTEE_DIR build directive is not
set. This change prevents build failures (see error trace below) with
some toolchains when build for product that do not provide OP-TEE OS
source tree path.

Building C object scp/CMakeFiles/scmi-fw.dir/config_all.o
cc1: error: /core/arch/arm/include: Not a directory
cc1: error: /core/include: Not a directory

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 427095866e1af55edb00ae8c06725dd002bd0458 module: Don't consider OP-TEE modules when SCP_OPTEE_DIR is not set]
Change-Id: I166d7646fca1431821030752ba3383c88659e8cc
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/404435
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 .../SCP-firmware/module/CMakeLists.txt           |  2 ++
 .../SCP-firmware/module/CMakeLists.txt.rej       | 16 ++++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt.rej

diff --git a/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt
index 96dacc947..0d3973385 100644
--- a/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt
+++ b/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt
@@ -44,12 +44,14 @@ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/mock_psu")
 #list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/mpmm")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/msg_smt")
 #list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/msys_rom")
+if(SCP_OPTEE_DIR)
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/clock")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/console")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/mbx")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/reset")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/smt")
 list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/voltd_regulator")
+endif(SCP_OPTEE_DIR)
 #list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pcid")
 #list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pik_clock")
 #list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pl011")
diff --git a/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt.rej b/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt.rej
new file mode 100644
index 000000000..2033d30cd
--- /dev/null
+++ b/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt.rej
@@ -0,0 +1,16 @@
+diff a/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt b/core/lib/scmi-server/SCP-firmware/module/CMakeLists.txt	(rejected hunks)
+@@ -44,12 +44,14 @@ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/mock_voltage_domain")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/mpmm")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/msg_smt")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/msys_rom")
++if(SCP_OPTEE_DIR)
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/clock")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/console")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/mbx")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/reset")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/smt")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/optee/voltd_regulator")
++endif(SCP_OPTEE_DIR)
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pcid")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pik_clock")
+ list(APPEND SCP_MODULE_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/pl011")
-- 
2.47.2

